fvm/src/engine.rs

```diff
@@ -557,8 +557,97 @@ impl Engine {
             .expect("failed to create available_gas global");
         store.data_mut().avail_gas_global = gg;
 
-        store.limiter(|id| id.kernel.limiter_mut());
+        fn as_wasmtime_limiter<K: Kernel>(
+            data: &mut InvocationData<K>,
+        ) -> &mut dyn wasmtime::ResourceLimiter {
+            unsafe {
+                &mut *(data.kernel.limiter_mut() as *mut _ as *mut WasmtimeLimiter<K::Limiter>)
+            }
+        }
+
+        store.limiter(as_wasmtime_limiter);
 
         store
     }
 }
+
+#[repr(transparent)]
+struct WasmtimeLimiter<L>(L);
```

Comment:
Couldn't this be moved into `limiter.rs`? It's quite substantial with its tests to be in `engine.rs` IMO.

It's the same crate, so I don't think an adapter to `wasmtime::ResourceLimiter` cannot be pushed to where we generally deal with limits.

Reply:
The difference is that _this_ is wasmtime specific, and the rest of the limiter isn't. We've tried to isolate wasmtime specific types as much as possible to:

- Make testing easier.
- Make upgrading wasmtime easier.
- Allow us to eventually support multiple wasm runtimes.